File: aspnetserver/Models/Prodotto.cs
using System;
using System.Collections.Generic;

namespace aspnetserver.Models;

public partial class Prodotto
{
    public int Id { get; set; }

    public string Nome { get; set; } = null!;

    public decimal? Prezzo { get; set; }

    public double? Peso { get; set; }

    public int? IdMarca { get; set; }

    public int? IdCategoria { get; set; }

    public int? IdLifeStyle { get; set; }

    public int? IdGalleria { get; set; }

    public int? IdPromozione { get; set; }

    public int? IdInformazione { get; set; }

    public virtual Marca? IdMarcaNavigation { get; set; }

    public virtual Promozione? IdPromozioneNavigation { get; set; }
}

File: react-shop/src/model/prodotto.ts
  export interface Prodotto {
    id: number;
    nome: string;
    prezzo: number;
    peso: number;
    idMarca?: number;
    idPromozione?: number;
    // idCategoria: number;
    // idLifeStyle: number;
    // idGalleria: number;
    // idInformazione: number;
    idMarcaNavigation: IdMarcaNavigation;
    idPromozioneNavigation: IdPromozioneNavigation;
  }

  export interface IdMarcaNavigation {
    id: number;
    nome: string;
    // prodotto?: (null)[] | null;
  }
  
  export interface IdPromozioneNavigation {
    id: number;
    nome: string;
    valore?: number;
    dataFine?: string;
    // prodotto?: (null)[] | null;
  }

  export type ProdottoFormCrea = Omit<Prodotto, 'id' | 'idMarcaNavigation' |'idPromozioneNavigation'>

  export type ProdottoFormModifica = Omit<Prodotto, 'idMarcaNavigation' |'idPromozioneNavigation'>
  